dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/UID.java

```diff
@@ -67,6 +68,7 @@ private UID(String value) {
   }
 
   @Override
+  @JsonValue
```

Comment:
> A typical usage is that of annotating toString() method so that returned String value is used as the JSON serialization; and if deserialization is needed, there is matching constructor or factory method annotated with [JsonCreator](https://fasterxml.github.io/jackson-annotations/javadoc/2.8/com/fasterxml/jackson/annotation/JsonCreator.html) annotation.
https://fasterxml.github.io/jackson-annotations/javadoc/2.8/com/fasterxml/jackson/annotation/JsonValue.html

Should we also add a `@JsonCreator` or how is a `UID` currently serialized?

Reply:
We have it here https://github.com/dhis2/dhis2-core/blob/2401fff481cfa8c8c31c2a1651348c3f9a231371/dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/UID.java#L78